java-manta-client/src/main/java/com/joyent/manta/client/multipart/EncryptedMultipartManager.java

```diff
@@ -396,6 +389,42 @@ private HttpContext buildRequestContext(final HttpContext context) {
     }
 
     /**
+     * Call wrapped.uploadPart() and reset encryption state in case of failure. This consists of recording encryption
+     * state using {@link EncryptionStateRecorder#record(EncryptionState, UUID)}, delegating to the {@code wrapped}
+     * manager and calling {@link EncryptionStateRecorder#rewind(EncryptionState, EncryptionStateSnapshot)} in case any
+     * errors occur. We expect the wrapped manager to validate the response code and throw a
+     * {@link MantaMultipartException} in case an unexpected response code is returned.
+     *
+     * @param upload          multipart upload object
+     * @param partNumber      part number to identify relative location in final file
+     * @param httpContext     additional request context, may be null
+     * @param encryptionState encryption state that should be reset in case of error
+     * @param entity          Apache HTTP Client entity instance
+     * @return the successfully uploaded part
+     * @throws IOException in case of network errors, though MantaMultipartException will be thrown in
+     *                     case of invalid response code
+     */
+    private MantaMultipartUploadPart uploadPartSafely(final EncryptedMultipartUpload<WRAPPED_UPLOAD> upload,
```

Comment:
Not sure why that name didn't pop into my head last night, I was too tired. Making this change.